cpufeatures/src/x86.rs

```diff
@@ -14,11 +14,11 @@ macro_rules! __unless_target_features {
     ($($tf:tt),+ => $body:expr ) => {{
         #[cfg(not(all($(target_feature=$tf,)*)))]
         {
-            #[cfg(not(target_env = "sgx"))]
+            #[cfg(not(any(target_env = "sgx", target_os = "none", target_os = "uefi")))]
             $body
 
-            // CPUID is not available on SGX targets
-            #[cfg(target_env = "sgx")]
+            // CPUID is not available on SGX, freestanding and UEFI targets
```

Comment:
This comment is a bit misleading. If I am not mistaken, CPUID is available on the later two targets, the issue is with compilation flags used by default on those targets which disable ability to use SIMD instructions. I wrote more about it here: https://internals.rust-lang.org/t/18169